include/nlohmann/detail/macro_scope.hpp

```diff
@@ -36,6 +36,10 @@
     #define JSON_HAS_CPP_11
 #endif
 
+#ifdef JSON_HAS_CPP_17
```

Comment:
Please do not move this include to this header. Rather, move it to all headers where `<optional>` is actually needed.